src/adRendering.js

```diff
@@ -253,8 +253,12 @@ export function renderAdDirect(doc, adId, options) {
  */
 export function insertLocatorFrame() {
   if (!window.frames[PB_LOCATOR]) {
-    const frame = createInvisibleIframe();
-    frame.name = PB_LOCATOR;
-    document.body.appendChild(frame);
+    if (!document.body) {
+      window.requestAnimationFrame(insertLocatorFrame);
```

Comment:
@olafbuitelaar on what assumption is this based, that this fixes the issue?

We use https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event to check if the DOM is ready. Does `requestAnimationFrame` imply something similar?

Reply:
That makes sense. Thanks for the clarification